### PR TITLE
v1.6.0 release note. [skip ci]

### PR DESCRIPTION
**Pull request by trivialfis**

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -3,6 +3,237 @@ XGBoost Change Log
...
+* Fix external memory with gpu_hist and subsampling. (#7481)
+* Check number of trees in inplace predict, this avoids a potential segfault when an
+  incorrect value for `iteration_range` is provided. (#7409)
+
+### Changes in the Python package
+Other than the changes in Dask, the XGBoost Python package gained some new features and
+improvements along with small bug fixes.
+
+* Python 3.7 is required as the lowest Python version. (#7682)
+* Binary package Support Apple Silicon. (#7621, #7612)
```
Comment (hcho3) [inline NEWS.md:128]:
```suggestion
* Pre-built binary wheel for Apple Silicon. (#7621, #7612) Apple Silicon users will now be able to run `pip install xgboost` to install XGBoost.
```

Can you also merge in https://github.com/dmlc/xgboost/pull/7747 as part of 1.6 release?

**Comment by wbo4958** [comment]
Could we wait for https://github.com/dmlc/xgboost/pull/7709 ?

**Comment by trivialfis** [comment]
https://github.com/dmlc/xgboost/pull/7759
https://github.com/dmlc/xgboost/pull/7752
https://github.com/dmlc/xgboost/pull/7756
https://github.com/dmlc/xgboost/pull/7759

File: NEWS.md [hunk 2]
```diff
@@ -3,6 +3,246 @@ XGBoost Change Log
...
+* During source build, XGBoost can choose cub in the system path automatically. (#7579)
+* XGBoost now honors the CPU counts from CFS, which is usually set in docker
+  environments. (#7654, #7704)
+* The metric `aucpr` is rewritten for better performance and GPU support. (#7297, #7368)
+* Metric calculation is now performed in double precision. (#7364)
+* XGBoost no longer mutates the global OpenMP thread limit. (#7537, #7519, #7608, #7590,
+  #7589, #7588, #7687)
+* The default behavior of `max_leave` and `max_depth` is now unified (#7302, #7551).
+* CUDA fat binary is now compressed. (#7601)
+* Deterministic result for evaluation metric and linear model In previous versions of
```
Comment (trivialfis) [inline NEWS.md:101]:
```suggestion
* Deterministic result for evaluation metric and linear model. In previous versions of
```

File: NEWS.md [hunk 3]
```diff
@@ -3,6 +3,250 @@ XGBoost Change Log
...
+rewrite closes the feature gap between `approx` and `hist` and improves the performance.
+Now the behavior and `approx` should be more aligned with `hist` and `gpu_hist`, here's a
+list of user-visible changes:
+
+- Supports both `max_leaves` and `max_depth`.
+- Supports `grow_policy`.
+- Supports monotonic constraint.
+- Supports feature weights.
+- Use `max_bin` to replace `sketch_eps`.
+- Supports for categorical data.
```
Comment (trivialfis) [inline NEWS.md:65]:
```suggestion
- Support for categorical data.
```

**Comment by wbo4958** [comment]
Please hold on, I would like to add some doc and samples for the GPU supporting in JVM package these days. @trivialfis

**Comment by wbo4958** [comment]
Hi @trivialfis, could you add this PR in 1.6.0 release https://github.com/dmlc/xgboost/pull/7784 thx

**Comment by wbo4958** [comment]
@trivialfis, could you add this PR in 1.6.0 release https://github.com/dmlc/xgboost/pull/7789

**Comment by hcho3** [comment]
Is this still WIP?

**Comment by trivialfis** [comment]
@hcho3 Changed the title.  It's complete.

File: NEWS.md [hunk 4]
```diff
@@ -3,6 +3,254 @@ XGBoost Change Log
...
+
+After a long period of development, XGBoost v1.6.0 is packed with many new features and
+improvements. We summarize them in the following sections starting with an introduction to
+some major new features, then moving on to language binding specific changes including new
+features and notable bug fixes for that binding.
+
+### Development of categorical data support
+This version of XGBoost features new improvements and full coverage of experimental
+categorical data support in Python and C package with tree model.  Both `hist`, `approx`
+and `gpu_hist` now supports training with categorical data.  Also, partition-based
```
Comment (hcho3) [inline NEWS.md:16]:
```suggestion
and `gpu_hist` now support training with categorical data.  Also, partition-based
```

File: NEWS.md [hunk 5]
```diff
@@ -3,6 +3,254 @@ XGBoost Change Log
...
+categorical split is featured in this release. This split type is first available in
```
Comment (hcho3) [inline NEWS.md:17]:
```suggestion
categorical split is introduced in this release. This split type is first available in
```

File: NEWS.md [hunk 6]
```diff
@@ -3,6 +3,254 @@ XGBoost Change Log
...
+LightGBM in the context of gradient boosting. In the previous version, only `gpu_hist`
+supports one-hot encoding-based split which has the form of `x \in {c}` where `{c}` is the
+set of all categories. In this new release, the `{c}` can be optionally split into 2 sets
+for the left and right nodes using any of the aforementioned tree methods. For more
```
Comment (hcho3) [inline NEWS.md:21]:
```suggestion
LightGBM in the context of gradient boosting. The previous XGBoost release supported one-hot split where the splitting criteria is of form `x \in {c}`, i.e. the categorical feature `x` is tested against a single candidate. The new release allows for more expressive conditions: `x \in S` where the categorical feature `x` is tested against multiple candidates. Moreover, it is now possible to use any tree algorithms (`hist`, `approx`, `gpu_hist`) when creating categorical splits. For more
```

File: NEWS.md [hunk 7]
```diff
@@ -3,6 +3,254 @@ XGBoost Change Log
...
+complete in XGBoost 1.6.  Building upon the iterator-based interface introduced in the
+previous version, now both `hist` and `approx` iterates over each batch of data during
+training and prediction.  In previous versions, `hist` concatenates all the batches into
+an internal representation, which is removed in this version.  As a result, users can
+expect higher scalability in terms of data size but might experience lower performance due
+to disk IO. (#7531, #7320, #7638, #7372)
+
+### Rewritten approx
+
+The `approx` tree method is rewritten based on the existing `hist` tree method, the
```
Comment (hcho3) [inline NEWS.md:55]:
```suggestion
The `approx` tree method is rewritten based on the existing `hist` tree method. The
```

File: NEWS.md [hunk 8]
```diff
@@ -3,6 +3,254 @@ XGBoost Change Log
...
+rewrite closes the feature gap between `approx` and `hist` and improves the performance.
+Now the behavior and `approx` should be more aligned with `hist` and `gpu_hist`, here's a
```
Comment (hcho3) [inline NEWS.md:57]:
```suggestion
Now the behavior of `approx` should be more aligned with `hist` and `gpu_hist`. Here is a
```

**Review by hcho3 — CHANGES_REQUESTED** [review]

**Review by hcho3 — APPROVED** [review]

**Pull request merged**